drivers/nfs.py

```diff
@@ -261,12 +263,13 @@ def get_supported_nfs_versions(server):
     valid_versions = set(['3', '4'])
     cv = set()
     try:
-        ns = util.pread2([RPCINFO_BIN, "-p", "%s" % server])
+        ns = util.pread2([RPCINFO_BIN, "-s", "%s" % server])
         ns = ns.split("\n")
         for i in range(len(ns)):
             if ns[i].find("nfs") > 0:
-                cvi = ns[i].split()[1]
-                cv.add(cvi)
+                cvi = ns[i].split()[1].split(",")
```

Comment:
It looks like this section gets no coverage in the tests, it probably didn't before either but given the newer, more complicated processing here it would be good to prove the code matches the intention by having some tests here. It's probably unlikely to get anything other than what is expected but if there are any error responses that need to be considered they should be tested as well.

Reply:
what happens here is that rpc info returns:
```
# rpcinfo -p <IP>
   program vers proto   port  service
    100000    4   tcp    111  portmapper
    100000    3   tcp    111  portmapper
    100000    2   tcp    111  portmapper
    100000    4   udp    111  portmapper
    100000    3   udp    111  portmapper
    100000    2   udp    111  portmapper
    100005    1   udp  56622  mountd
    100005    1   tcp  42377  mountd
    100005    2   udp  36412  mountd
    100005    2   tcp  60855  mountd
    100005    3   udp  37001  mountd
    100005    3   tcp  53917  mountd
    100003    3   tcp   2049  nfs
    100003    4   tcp   2049  nfs
    100227    3   tcp   2049  nfs_acl
    100003    3   udp   2049  nfs
    100227    3   udp   2049  nfs_acl
    100021    1   udp  35853  nlockmgr
    100021    3   udp  35853  nlockmgr
    100021    4   udp  35853  nlockmgr
    100021    1   tcp  39059  nlockmgr
    100021    3   tcp  39059  nlockmgr
    100021    4   tcp  39059  nlockmgr

# rpcinfo -s <IP>
   program version(s) netid(s)                         service     owner
    100000  2,3,4     local,udp,tcp,udp6,tcp6          portmapper  superuser
    100005  3,2,1     tcp6,udp6,tcp,udp                mountd      superuser
    100003  4,3       udp6,tcp6,udp,tcp                nfs         superuser
    100227  3         udp6,tcp6,udp,tcp                nfs_acl     superuser
    100021  4,3,1     tcp6,udp6,tcp,udp                nlockmgr    superuser
```

But how can i test it without a real target?

Reply:
Use the mocks in the unit tests. As an example - https://github.com/xapi-project/sm/blob/master/tests/test_nfs.py#L40 which provides the data for the return of the command